sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/ParquetVectorUpdaterFactory.java

```diff
@@ -179,6 +183,21 @@ boolean isTimestampTypeMatched(LogicalTypeAnnotation.TimeUnit unit) {
       ((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).getUnit() == unit;
   }
 
+  void validateTimestampType(DataType sparkType) {
+    assert(logicalTypeAnnotation instanceof TimestampLogicalTypeAnnotation);
+    // Throw an exception if the Parquet type is TimestampLTZ and the Catalyst type is TimestampNTZ.
+    // This is to avoid mistakes in reading the timestamp values.
+    if (((TimestampLogicalTypeAnnotation) logicalTypeAnnotation).isAdjustedToUTC() &&
+      sparkType == DataTypes.TimestampNTZType) {
+      convertErrorForTimestampNTZ("int64 time(" + logicalTypeAnnotation + ")");
+    }
+  }
+
+  void convertErrorForTimestampNTZ(String parquetType) {
+    throw new RuntimeException("Unable to create Parquet converter for data type " +
+      DataTypes.TimestampNTZType.json() + " whose Parquet type is " + parquetType);
```

Comment:
can we use the SQL format? `TIMESTAMP WITHOUT TIMEZONE`

Reply:
Both are "timestamp_ntz"

Reply:
I was not referring to `.sql`, I mean "TIMESTAMP WITHOUT TIMEZONE".

Reply:
I prefer keeping "timestamp_ntz", which is the keyword we used in DDL/literals..

Reply:
nvm, `timestamp_ntz` is kind of an alias of `TIMESTAMP WITHOUT TIMEZONE` in Spark today, and a shorter name is better here.